src/main.cpp

```diff
@@ -895,12 +890,22 @@ bool ContextualCheckTransaction(
                 REJECT_INVALID, "bad-tx-sapling-version-too-low");
         }
 
-        // Reject transactions with invalid version
-        if (tx.nVersion > SAPLING_MAX_TX_VERSION) {
-            return state.DoS(
-                dosLevelPotentiallyRelaxing,
-                error("ContextualCheckTransaction(): Sapling version too high"),
-                REJECT_INVALID, "bad-tx-sapling-version-too-high");
+        if (!futureActive) {
```

Comment:
Why are these checks duplicated into the `futuresActive` checks?

Reply:
The alternative of how these checks could be performed is this: https://github.com/zcash/zcash/commit/2827cb2c169c20008adc712296aedb4c9e040ded

I don't much care for this though; it's harder to see what's going on.